libc/src/__support/str_to_float.h

```diff
@@ -1055,6 +1055,38 @@ hexadecimal_string_to_float(const char *__restrict src,
   return output;
 }
 
+template <class T>
+LIBC_INLINE fputil::FPBits<T> nan_from_ncharseq(const char *__restrict ncharseq,
```

Comment:
This is mostly fine, but we can simplify this function prototype a bit more.

This function doesn't actually do anything based on the sign except set it in the result, so that can be removed here and handled in strtofloatingpoint.

The ncharseq with length can be a single string_view variable (see `src/__support/CPP/string_view.h`).

The error can be part of a struct return type, similar to `str_to_num_result`.

The actual returned value doesn't need to be an FPBits object, if we move the conversion to float elsewhere this function only needs to return a `uint64_t` (and the error).

Reply:
> The actual returned value doesn't need to be an FPBits object, if we move the conversion to float elsewhere this function only needs to return a uint64_t (and the error).

By `uint64_t` you mean the nan_mantissa?

> The error can be part of a struct return type, similar to str_to_num_result.
Just out of curiosity, what benefit does it have to make it a part of return payload. Won't it be a little faster/efficient to assign it directly to the `error` object we'll be using?

Reply:
Yes I mean the NaN mantissa, but it's just being cast from the static `uint64_t` so we can let the caller handle that and avoid templating this function altogether.

On lower optimization levels the pointer vs the struct might make a difference, but in any practical configuration they'll be effectively identical from a performance perspective. Here's an example on godbolt so you can see the difference: https://godbolt.org/z/oMqb3r6j6
In general we try to avoid using pointers to return values because it's very error prone. It's easy to miss a value being modified if it's being passed as a pointer, especially if it's only modified sometimes.

Reply:
Makes sense, thank you for the explanation. Done!